tests/lib/reset.sh

```diff
@@ -86,6 +87,9 @@ reset_classic() {
 
         # force all profiles to be re-generated
         rm -f /var/lib/snapd/system-key
+
+        # force snapd-session-agent.socket fto be re-generated
+        rm -f /run/user/0/snapd-session-agent.socket
```

Comment:
Perhaps we should also have `systemctl --user stop snapd-session-agent.socket || true` somewhere after we stop snapd.service. Note this may need to run on systems where the user session systemd is support so something like:
```
if [ -S /run/user/0/snapd-session-agent.socket ] ; then
    systemctl --user stop snapd-session-agent.socket
fi
```